### DISCUSSION TITLE
How to change the output format?

### DISCUSSION THREAD

**Discussion by lucky33newman** [pytest-dev/pytest#10455] [Q&A] [answered]
This is like this:

root@raspberrypi4-64:/usr/AGL/agl-test/tests# agl-test -vs
===================================================================================================================================== test session starts =====================================================================================================================================
platform linux -- Python 3.10.4, pytest-7.1.1, pluggy-1.0.0 -- /usr/bin/python3
cachedir: /var/run/agl-test/.pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/usr/AGL/agl-test/.hypothesis/examples')
rootdir: /usr/AGL/agl-test, configfile: pytest.ini
plugins: hypothesis-6.39.5
collected 153 items

tests/libpam/run_tests.py::test_name[tst-pam_acct_mgmt] PASSED
tests/libpam/run_tests.py::test_name[tst-pam_authenticate] PASSED
tests/libpam/run_tests.py::test_name[tst-pam_chauthtok] PASSED
tests/libpam/run_tests.py::test_name[tst-pam_close_session] PASSED
tests/libpam/run_tests.py::test_name[tst-pam_end] PASSED
tests/libpam/run_tests.py::test_name[tst-pam_fail_delay] PASSED

balabala......
++++++++++++++++++++++++++++++++++++++++++
As you can see, I am getting output format like:
tests/libpam/run_tests.py::test_name[tst-pam_fail_delay] PASSED

Question is, how can I decide the name of this part of the output:

tests/libpam/run_tests.py

For example I would perfer it to just be like: 

libpam_tests
++++++++++++++++++++++++++++++++++++++++++
My run test script as below:

import pytest

import tests.libpam.parser as parser

from plugins.agl_test_ptest_base import PTESTBase

class LIBPAMBase(PTESTBase):
    def __init__(self):
        super().__init__(test_name="libpam")

    def log_process(self):
        log_file = self.get_logfile()
        self.case_info_list = parser.log_parse(log_file)
        global global_case_name_list
        global_case_name_list = self.case_info_list.keys()
        self.init_case_status()

    def paramtrize_init(self):
        if(self.precheck()==True):
            self.run_test_fun()
            self.log_process()
        else:
            global global_case_name_list
            # empty parametrize list, skip automatically
            global_case_name_list = []

instance = LIBPAMBase()
instance.paramtrize_init()

@pytest.fixture(scope='module')
def testbase():
    yield instance
    #package log files and make report file
    instance.log_report()

def setup_module():
    pass

@pytest.mark.parametrize(
    'case_name',
    global_case_name_list
)

def test_name(testbase: LIBPAMBase, case_name):
    testbase.run_case(case_name)

if __name__ == '__main__':
    pytest.main("-s run_tests")

**Comment by nicoddemus** [comment, marked as answer]
(If possible please try to format your post correctly)

That is not currently customizable, if you *really* want to do it, you can do it by writing your own plugin. See `pytest-sugar` for an example.

**Comment by lucky33newman** [comment]
Seems inavidable to use the sugar.

Thanks.
Yan

Bruno Oliveira ***@***.***> 于2022年11月2日周三 21:34写道：
